mantidproject/mantid
# Supress many CMake warnings: Use FindPython over deprecated FindPythonInterp and FindPythonLibs

**Description of work.**

Moves to CMake FindPython and sets the minimum Python version to 3.6.

Fixes various cmake warnings that are confusing to developers. One remains on some platforms regarding missing headers for boost python but this can be ignored an cannot be silenced.

Also reduces the amount of times various finders are called for the same package.

This is the first step in trying got move to finders with imported targets.

**To test:**

Code review.
Run cmake and see that Qt4/Qt5 and Boost Python are only found once.
On Windows cmake in the GUI should produce no red warning text.

Fixes #16997
Refs #27722 

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
